modules/porous_flow/include/materials/PorousFlowMaterialPermeabilityFromPorosity.h

```diff
@@ -0,0 +1,76 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef POROUSFLOWMATERIALPERMEABILITYFROMPOROSITY_H
+#define POROUSFLOWMATERIALPERMEABILITYFROMPOROSITY_H
+
+#include "DerivativeMaterialInterface.h"
+#include "Material.h"
+
+#include "PorousFlowDictator.h"
+
+//Forward Declarations
+class PorousFlowMaterialPermeabilityFromPorosity;
+
+template<>
+InputParameters validParams<PorousFlowMaterialPermeabilityFromPorosity>();
+
+/**
+ * Material designed to provide the permeability tensor
+ * which is a function of porosity
```

Comment:
Here you should explain the functional permeability = porosity^n/(1 - porosity)^m, etc